pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -409,7 +426,7 @@ func (c *Controller) queueServiceForEndpointSlice(endpointSlice *discovery.Endpo
 		utilruntime.HandleError(fmt.Errorf("Couldn't get key for EndpointSlice %+v: %v", endpointSlice, err))
 		return
 	}
-	c.queue.Add(key)
+	c.queue.AddRateLimited(key)
```

Comment:
I'm not sure I'm really following this change.
Basically, with this change what you're saying is that we start rate-limiting resyncing service on endpoint-slice changes.
But what does it buy us?
- if the service is already queued, rate-limiting only can make things worse, because the state based on which we do resync (cache) is already up-to-date, so another resync will be no-op anyway
- if the service isn't queued, then it should have be Forget previously, which is zero-ing ratelimiter for that item: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/util/workqueue/rate_limiting_queue.go#L68

So I think this can only make things worse - what am I missing?

Reply:
Agree... we chatted about this offline. Switching this to `AddAfter` with a batch time period could make sense (to reduce the number of service queues in response to a bunch of endpoint slice changes).

I would prefer to keep the AddRateLimited calls focused on error requeue cases.

Reply:
You're right, thanks. This is now using `AddAfter()` with a constant 1 second delay.